### PR TITLE
[RPA-958] SharePoint excel action rework: Get cell

### PR DESCRIPTION
**Pull request by bhszewczyk**
## UI update:

1. Changed action name from "Get content from cell" to "Get cell"
2. Added tooltip with proper input example
3. Changed output to be optional

image

## Functionality:
1. Previously: action was working only when column name included only one letter and row number was below 10, now action works for every cell address
2. Previously: output was a MSGraphService object with all unnecessary information included, now outpur is a single value
3. Output is a cell value (with type kept) with date and percentage exception, where the output is returned as string, examples:

- text:
image
image

- number:
image
image

- percetage:
image
image

- date:
image
image

- currency:
image
image

4. Adds incorrect input error:
image
image

## Motivation and Context

- action name was too long
- action didn't work for all cell addresses
- there is no information what we exect from the user

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 1]
```diff
@@ -71,11 +71,16 @@ export class ExcelService {
...
+        try {
+            const response = this.microsoftGraphService.get(this.createWorkbookUrl(url), {
+                headers: {
+                    'workbook-session-id': this.session.workbookSessionInfo.id,
+                },
+            });
+            return response;
+        } catch(error) {
+            throw new Error(error.message);
+        }
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: redundant try clause

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 2]
```diff
@@ -142,7 +147,8 @@ export class ExcelService {
         // it uses the ASCII code of the letter to calculate the number
         // e.g A = 65, B = 66, C = 67, etc.
         // so we subtract 64 from the ASCII code to get the column number
-        return (column.length - 1) * 26 + (column.charCodeAt(column.length - 1) - 64);
+        const colnum = (column.length - 1) * 26 + (column.charCodeAt(column.length - 1) - 64);
+        return colnum;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: if we need to create a variable here then name it more clearly, avoid abbreviations

File: runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts [hunk 3]
```diff
@@ -119,7 +119,7 @@ export class DesktopRunnerService implements OnModuleInit {
         let currentExtensionName: string;
         try {
             const extensions = readdirSync(this.serverConfigService.extensionsDirPath, { withFileTypes: true })
-                .filter((dirent) => dirent.isDirectory())
+                .filter((dirent) => dirent.isDirectory() && dirent.name !== 'common' && dirent.name[0] !== '.')
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts]: typo `dirent`

Comment (preludium) [reply runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts]: maybe use `startsWith` instead of 0 index

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts (hunk 1 above)
Comment (bhszewczyk) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: removed

File: runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts (hunk 3 above)
Comment (bhszewczyk) [reply runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts]: what typo? 🤔 
image
image

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts (hunk 2 above)
Comment (bhszewczyk) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: no need for variable creation, forgot to remove it

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 4]
```diff
@@ -67,15 +67,16 @@ export class ExcelService {
...
 
-        return this.microsoftGraphService.get(this.createWorkbookUrl(url), {
+        const response = this.microsoftGraphService.get(this.createWorkbookUrl(url), {
             headers: {
-                'workbook-session-id': this.session.workbookSessionInfo.id,
-            },
+                'workbook-session-id': this.session.workbookSessionInfo.id
+            }
         });
+        return response;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: please, just return ms graph usage

File: runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts [hunk 5]
```diff
@@ -119,8 +117,8 @@ export class DesktopRunnerService implements OnModuleInit {
         let currentExtensionName: string;
         try {
             const extensions = readdirSync(this.serverConfigService.extensionsDirPath, { withFileTypes: true })
-                .filter((dirent) => dirent.isDirectory())
-                .map((dirent) => dirent.name);
+                .filter(dirent => this.detectExtensionsDirectories(dirent))
+                .map(dirent => dirent.name);
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts]: fix the typo plx `dirent`

File: runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts [hunk 6]
```diff
@@ -240,4 +239,26 @@ export class DesktopRunnerService implements OnModuleInit {
...
+
+    private detectExtensionsDirectories(dirent: Dirent) {
+        return (
+            dirent.isDirectory() &&
+            !dirent.name.startsWith('.') &&
+            this.checkIfFileSystemEntryPresentInDirectory(dirent.path, dirent.name, [
+                'dist',
+                'node_modules',
+                'package.json',
+                'package-lock.json'
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts]: there are 3 options each for every package manager `package-lock.json` for npm, `yarn.lock` for yarn and `pnpm-lock.yaml` for pnpm

**Review by preludium — CHANGES_REQUESTED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**